### DISCUSSION TITLE
Multiple provider linking support?

### DISCUSSION THREAD

**Discussion by IDrumsey** [nextauthjs/next-auth#5548] [Help]
### Question 💬

I'm using the Firebase adapter and in my Firebase project's authentication settings I am using the `create multiple accounts for each identity provider` option instead of the `link accounts that use the same email` option. After trying to sign in with next-auth with an email that is already signed up, but with a different provider, I keep getting the AccountNotLinkedError exception seen below. Does next-auth support creating multiple accounts with the same email address for different providers? Below says, "We don't want to have two accounts with the same email address". What is the reason behind this? If it's not a security concern, is there a way around this?

https://github.com/nextauthjs/next-auth/blob/eb5a9bad9d69e228fba173fd29b2e96bc9154453/packages/next-auth/src/core/lib/callback-handler.ts#L182-L192

### How to reproduce ☕️

firebase adapter wrapping multiple at least 2 providers.
Firebase project -> authentication -> settings -> user account linking -> `create multiple accounts for each identity provider`

Try signing in with provider 1. Should be successful.
Sign out.
Try signing in with provider 2 with the same email as provider 1. Should throw the `AccountNotLinkedError` exception.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by nbouvrette** [comment]
I'm surprised nobody replied back but did you try setting `allowDangerousEmailAccountLinking: true` on your [providers](https://next-auth.js.org/configuration/providers/oauth#allowdangerousemailaccountlinking-option)?

As you said the main concern is security, so by default providers are not trusted. Of course, if you trust well-known providers like Google, Github, etc, it won't cause any security risk since they already have robust processes on their end.
